src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,780 @@
+                {
+                    TargetMethod.Name: nameof(Equals),
+                    Instance: IOperation instance,
+                    Arguments: [IArgumentOperation { Value: IOperation arg }]
+                })
+            {
+                return false;
+            }
+
+            // get the (potential) parameters, uwrapping any potential implicit casts
+            var invokedOn = GetParamFromArgument(instance);
+            var param = GetParamFromArgument(arg);
+            // make sure we're not referencing the same parameter twice
+            return param != null && invokedOn != null && !invokedOn.Equals(param);
+        }
+
+        /// <summary>
+        /// Get the referenced parameter (and unwraps implicit cast if necessary) or null if a parameter wasn't referenced
+        /// </summary>
+        /// <param name="arg">The operation for which to get the parameter</param>
+        /// <returns>the referenced parameter or null if unable to find</returns>
+        private static IParameterSymbol? GetParamFromArgument(IOperation arg)
+        {
+            var bottom = arg.WalkDownConversion();
+            if (bottom is IParameterReferenceOperation parameterReference)
+            {
+                return parameterReference.Parameter;
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Matches constructors where each statement simply assigns one of the provided parameters to one of the provided properties
+        /// with no duplicate assignment or any other type of statement
+        /// </summary>
+        /// <param name="constructor">Constructor Declaration</param>
+        /// <param name="properties">Properties expected to be assigned (would be replaced with positional constructor)</param>
+        /// <param name="parameters">Constructor parameters</param>
+        /// <returns>Whether the constructor body matches the pattern described</returns>
+        private static bool IsSimpleConstructor(
+            ConstructorDeclarationSyntax constructor,
+            ImmutableArray<PropertyDeclarationSyntax> properties,
+            SeparatedSyntaxList<ParameterSyntax> parameters,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            if (semanticModel.GetOperation(constructor, cancellationToken) is not IConstructorBodyOperation operation)
+            {
+                return false;
+            }
+
+            var body = operation.BlockBody ?? operation.ExpressionBody;
+
+            // We expect the constructor to have exactly one statement per parameter,
+            // where the statement is a simple assignment from the parameter to the property
+            if (body == null || body.Operations.Length != parameters.Count)
+            {
+                return false;
+            }
+
+            var propertyNames = properties.SelectAsArray(p => p.Identifier.ToString());
```

Comment:
Does this need to be using ValueText or will this get tripped up with names that were escaped but didn't have to be?